.github/workflows/release.yml

```diff
@@ -63,3 +63,178 @@ jobs:
+
+      - name: "Copy GAP's release scripts to a safe place"
+        shell: bash
+        run: cp -rp ${GAPDEV_DIR}/dev/releases .
+
+      - uses: actions/setup-python@v2
+        if: ${{ startsWith(github.ref, 'refs/tags/v') }}
+
+      - name: "Install required Python modules"
+        if: ${{ startsWith(github.ref, 'refs/tags/v') }}
+        run: pip3 install PyGithub
+
+      # We could probably use either a GitHub Action from the GitHub Marketplace
+      # to download the appropriate GAP tarball from the GitHub release, rather
+      # than creating and using the Python file.  This would probably mean that
+      # we lose the ability to verify checksums, but that is probably fine.
+      - name: "Download the appropriate GAP release tarball"
+        if: ${{ startsWith(github.ref, 'refs/tags/v') }}
+        shell: bash
+        run: |
+          python -u ./releases/download_release_archive.py v${VERSION} gap-${VERSION}.tar.gz .
+          tar -zxf gap-${VERSION}.tar.gz
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      # TODO When the PR https://github.com/ChrisJefferson/sage-windows/pull/1
+      # is merged, then should be updated to ChrisJefferson/sage-windows.
+      # Ultimately, this repository should be owned by one of the GAP-related
+      # organisations, and perhaps ultimately it can be re-integrated with
+      # the original sagemath/sage-windows repository.
+      - name: "Clone the Windows installer maker scripts"
+        run: git clone --branch master https://github.com/wilfwilson/sage-windows.git
+
+      - uses: gap-actions/setup-cygwin-for-gap@v1
+
+      # TODO Investigate how to speed this up. For instance, if we do not need
+      # to compile the GAP manuals (e.g. if starting with an actual release
+      # tarball), then perhaps we can do without installing latex?
+      #
+      # Currently, the sage-windows/release_gap.sh script wraps the GAP
+      # contained in gap-$VERSION, and outputs its installer to
+      # sage-windows/Output/gap-${{ env.VERSION }}-${{ matrix.arch }}.exe
+      #
+      # "CHERE_INVOKING=1 lets us start a 'login shell' (to set paths) without changing directory"
+      - name: "Compile GAP and its packages, and create the installer"
+        shell: cmd
+        run: |
+          @ECHO ON
+          SET CHERE_INVOKING=1
+          cd sage-windows
+          C:\cygwin64\bin\bash -l -c 'bash release_gap.sh'
+        env:
+          ARCH: ${{ matrix.arch }}
+          SAGE_BUILD_DOC_CMD: '"true"'
+          SAGE_VERSION: ${{ env.VERSION }}
+
+      # TODO Decide the situation in which one might wish to upload the
+      # installer as an artifact.
+      - name: "Upload the installer as an artifact"
+        if: ${{ !startsWith(github.ref, 'refs/tags/v') }}
+        uses: actions/upload-artifact@v2
+        with:
+          name: gap-${{ env.VERSION }}-${{ matrix.arch}}.exe
+          path: sage-windows/Output/gap-${{ env.VERSION }}-${{ matrix.arch }}.exe
```

Comment:
Since the artifacts are huge and may stay around for a bit, I say let's not upload artifacts for every commit and every PR. 

If we want to run this workflow on every commit and PR, I'd say let's never upload artifacts. Or let's only upload them on failure, but then there probably won't be anything to upload anyways. If there are failures, there always is the "debugging via tmate" action.

If we decide to only run the full workflow when triggered manually, then in that case uploading the artifacts should be fine IMO.

Reply:
FYI, you can specify how long to keep artifacts, with a minimum of 1 day: https://github.com/actions/upload-artifact#retention-period – this might be useful on some/all of our workflows; see also https://docs.github.com/en/actions/reference/usage-limits-billing-and-administration#artifact-and-log-retention-policy

Reply:
Note that even the logs of this workflow are BIG - 50MB or so, I think.

Reply:
Ooh nice. So then instead of not uploading artifacts I'm for keeping them around for one day.